public/components/CampaignAnalytics/Analytics/CampaignReferrals.js

```diff
@@ -5,60 +5,77 @@ import {getCampaignReferrals, setToggleNode} from "../../../actions/CampaignActi
 import CampaignReferralHeader from "./CampaignReferralHeader";
 import InfinityMenu from "react-infinity-menu";
 import ProgressSpinner from "../../utils/ProgressSpinner";
+import DateRangeEditor from "./DateRangeEditor";
 
 class CampaignReferrals extends React.Component {
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      dateRange: {
+        startDate: DateRangeEditor.toDate(this.props.campaign.startDate),
+        endDate: DateRangeEditor.toDate(this.props.campaign.endDate)
+      }
+    };
+  }
+
   componentWillMount() {
-    this.props.campaignReferralActions.getCampaignReferrals(this.props.campaign.id);
+    this.props.campaignReferralActions.getCampaignReferrals(this.props.campaign.id, this.state.dateRange);
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.campaign.id !== this.props.campaign.id) {
-      this.props.campaignReferralActions.getCampaignReferrals(nextProps.campaign.id);
+  componentWillUpdate(nextProps, nextState) {
+    if (nextProps.campaign.id !== this.props.campaign.id || nextState !== this.state) {
```

Comment:
I'm not sure if that test really captures specific-case update criteria. Is this supposed to detect changes in `dateRange`? could it be more specific? also, the params could just be an object, that encapsulates all the state that parameterises the query (the campaign id, date range). It's also a bit confusing having redux and `componentWillUpdate` overlapping each other.

Reply:
@rich-nguyen Is this clearer?  I don't think the change in properties is actually relevant so it's just a particular change in state that I need to look out for.